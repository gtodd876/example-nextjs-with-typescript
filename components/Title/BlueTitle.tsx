import styled from 'styled-components';
import { Title } from './Title';

export const BlueTitle = styled(Title)`
  color: blue;
`;
