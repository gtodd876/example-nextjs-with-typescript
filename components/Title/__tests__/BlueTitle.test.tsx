import React from 'react';
import { render } from '@testing-library/react';
import { BlueTitle } from '../BlueTitle';

describe('BlueTitle', () => {
  test('Styled BlueTitle component', () => {
    const { container } = render(<BlueTitle>Test</BlueTitle>);
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('font-size', '50px');
    expect(container.firstChild).toHaveStyleRule('color', 'blue');
  });
});
