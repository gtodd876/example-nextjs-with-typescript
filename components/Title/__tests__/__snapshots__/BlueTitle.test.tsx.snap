// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlueTitle Styled BlueTitle component 1`] = `
.c0 {
  font-size: 50px;
  color: blue;
}

<div>
  <h1
    class="c0"
  >
    Test
  </h1>
</div>
`;
